src/main.ts

```diff
@@ -113,18 +114,42 @@ async function enterDungeon(level: number): Promise<void> {
   state.player.sprite.visible = true;
   centerCameraOn(state.camera, state.player.sprite, state.app.screen);
 
+  state.world.board = updateTilesVisibility(state.player, state.world.board);
   state.world.enemies = spawnEnemies(level * 2 + 4);
   state.world.entities = spawnEntities(state.world);
-  state.world.entities = updateEntitiesVisibility(state.world.entities, state.player);
-  state.world.board = updateTilesVisibility(state.player, state.world.board);
+  state.world.entities = updateEntitiesVisibility(state.world.entities);
+
+  let moveResult = MoveResult.None;
+
+  /* eslint-disable no-await-in-loop */
+  while (isAutoMovement) {
+    await timeout(5000 / state.player.speed);
+
+    moveResult = await movePlayerToCell(selectNextMove(state.player, state.world.board));
+
+    if (moveResult !== MoveResult.None) {
+      break;
+    }
+
+    // basic enemies moving
+    // TODO: avoid moving to the player cell or starting a combat
```

Comment:
We actually want the enemies to engage the player sometimes, though I'm pretty sure our current logic doesn't support that yet.